### PR TITLE
SLE 2024-09 release notes 

### PR DESCRIPTION
**Pull request by CIakab-NI**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/install-systemlink-enterprise/blob/master/CONTRIBUTING.md).

### What does this Pull Request accomplish?

Adding SLE 2024-09 release notes

### Why should this Pull Request be merged?

Required for the 2024-09 releases

### What testing has been done?

N/A

### CODE REVIEW

**Comment by CIakab-NI** [comment]
@prestwick , @vpstoynova , could you help review the sections:
- New Features and Behavior changes
- Helm Chart Breaking Changes

The document that includes this information from the squads is this one:
- [NI SharePoint: SLE 2024-09_Helm Chart Breaking Changes, Software Behavior Changes (including Privileges) for the Release Notes - out of August 2024.xlsx](https://nio365.sharepoint.com/:x:/s/ASWSystemLink/EYBWHBYNwAVNqnhDnG-5v70BC1tPvPW3xUhO2Qb7d-NVyg?e=NeviZu)

Pending information to be added to the Readme's New Features and Behavior changes:
- New Features as documented by Vera and the SL POs in [systemlink-enterprise-new-features-and-changes.md](https://dev.azure.com/ni/Users/_git/Documentation%20Reviews?path=%2Freview%20output%2Fsystemlink-enterprise%20md%2FSystemLink%20Enterprise%20New%20Features%20and%20Changes%2Fsystemlink-enterprise-new-features-and-changes.md&_a=preview)

File: release-notes/2024-09/README.md [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+
+## Software Bill of Materials and Notices
+
+[SBOM](https://github.com/ni/install-systemlink-enterprise/tree/2024-09/release-notes/2024-09/sbom)
+
+[Notices](https://github.com/ni/install-systemlink-enterprise/tree/2024-09/release-notes/2024-09/notices)
+
+## Versions
+
+**Top Level Helm Chart:** `systemlink 0.29.49`
```
Comment (darrenbiel) [inline release-notes/2024-09/README.md:60]: I'll need to bump these versions once the cherry-pick bugs are in.

File: release-notes/2024-09/README.md [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+
+## New Features and Behavior changes
+
+- The Alarm Service automatically deletes active alarms that haven't been updated in a configurable number of days.
+    - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L469)
+- The Alarm Service enforces configurable limits on the number of active alarms that can be created, as well as the total number of alarms that can be created.
+    - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L484)
+- The Schedule page has been added under Operations in the Navigation Pane. The users can now view all the scheduled Test Plans, across all Systems in the Lab, in a timeline view.
```
Comment (vpstoynova) [inline release-notes/2024-09/README.md:11]:
```suggestion
- The Schedule page was added under Operations in the Navigation Pane. You can now view all the scheduled test plans, across all systems, in a timeline view.
```
```suggestion
- The Schedule page has been added under Operations in the Navigation Pane. The users can now view all the scheduled Test Plans, across all Systems in the Lab, in a timeline view.
```

File: release-notes/2024-09/README.md [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+- Privilege **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment** was split to **List available notebooks** to list the published notebooks, and **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment** to list and download notebooks for notebook development.
+    - This privilege is enabled by default for Collaborator, Data Maintainer and Systems Maintainer built-in roles.
```
Comment (vpstoynova) [inline release-notes/2024-09/README.md:13]:
```suggestion
    - This privilege is enabled by default for Collaborator, Data Maintainer, and Systems Maintainer built-in roles.
```
```suggestion
    - This privilege is enabled by default for Collaborator, Data Maintainer and Systems Maintainer built-in roles.
```

File: release-notes/2024-09/README.md [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+    - For the Automated Agent, the **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment** privilege will be removed, as the Automated Agent should not need to download the notebooks content.
+- Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on Jupyter user servers to be used when calling SystemLink or other external APIs from notebooks.
```
Comment (vpstoynova) [inline release-notes/2024-09/README.md:15]:
```suggestion
- Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on servers that use Jupyter. These certificates are used when calling SystemLink or other external APIs from notebooks.
```
```suggestion
- Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on Jupyter user servers to be used when calling SystemLink or other external APIs from notebooks.
```

File: release-notes/2024-09/README.md [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+- Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on execution pods, to be used when calling SystemLink or other external APIs from notebooks.
```
Comment (vpstoynova) [inline release-notes/2024-09/README.md:16]:
```suggestion
- Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on execution pods. These certificates are used when calling SystemLink or other external APIs from the pods.
```
```suggestion
- Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on execution pods, to be used when calling SystemLink or other external APIs from notebooks.
```

**Review by vpstoynova — APPROVED** [review]

File: release-notes/2024-09/README.md [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+    - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L469)
+- Adjust how many alarms you can create in the SystemLink Enterprise Alarm Service - [Configuring Alarm Limits](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/configuring-alarm-limits.html).
+    - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L484)
+- Learn about the [performance metrics for the Dashboard Host Alarm Service](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/performance-metrics-dashboard-host-service.html).
+- Privilege **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment** was split to **List available notebooks** to list the published notebooks, and **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment** to list and download notebooks for notebook development.
```
Comment (prestwick) [inline release-notes/2024-09/README.md:16]:
```suggestion
- The privilege **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment** was split to **List available notebooks** and **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment**.
```

File: release-notes/2024-09/README.md [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+    - This privilege is enabled by default for Collaborator, Data Maintainer, and Systems Maintainer built-in roles.
```
Comment (prestwick) [inline release-notes/2024-09/README.md:17]:
```suggestion
    - These privileges are enabled by default for Collaborator, Data Maintainer, and Systems Maintainer built-in roles.
```

File: release-notes/2024-09/README.md [hunk 8]
```diff
@@ -0,0 +1,221 @@
...
+    - For the Automated Agent, the **List published notebooks in SystemLink and download published notebooks into the Jupyter development environment** privilege will be removed, as the Automated Agent should not need to download the notebooks content.
+- Support for [Private Certificate Authorities](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/private-certificate-authorities.html).
+    - Connect to MongoDB and S3 compatible interfaces that use certificates signed by a private authority.
+    - Call SystemLink APIs from Jupyter Notebook when the SystemLink API endpoint is configured to use a certificate signed by a private authority.
+    - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on servers that use Jupyter. These certificates are used when calling SystemLink or other external APIs from notebooks.
```
Comment (prestwick) [inline release-notes/2024-09/README.md:22]:
```suggestion
    - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on containers that run Jupyter notebooks. These certificates are used when calling SystemLink or other external APIs from notebooks.
```

File: release-notes/2024-09/README.md [hunk 9]
```diff
@@ -0,0 +1,221 @@
...
+    - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on execution pods. These certificates are used when calling SystemLink or other external APIs from the pods.
```
Comment (prestwick) [inline release-notes/2024-09/README.md:23]: With changes suggested in the previous line we can remove this line because it will be worded to include both Jupyter user pods and notebook execution workers running Jupyter notebooks.

File: release-notes/2024-09/README.md [hunk 10]
```diff
@@ -0,0 +1,221 @@
...
+- Added abilities to configure the rate-limit values for notebook APIs inside the Helm chart.
```
Comment (prestwick) [inline release-notes/2024-09/README.md:24]: We need a reference to where and how this is done. This likely will require us to update the system link-value.yaml in this repo to expose this configuration. Alternatively we could reference the ni.com doc on this configuration should one exist.

File: release-notes/2024-09/README.md [hunk 11]
```diff
@@ -0,0 +1,221 @@
...
+- Notebook Execution Service
+    - Created a dedicated swagger page for Artifact APIs.
+    - Implemented Rate limiting for the Artifact APIs.
+    - Created S3 buckets for Artifact APIs.
+- Notebook Execution Service, Systems Management Service
```
Comment (prestwick) [inline release-notes/2024-09/README.md:29]:
```suggestion
- Systems Management Service
```

File: release-notes/2024-09/README.md [hunk 12]
```diff
@@ -0,0 +1,221 @@
...
+    - Added support for deploying a custom CA root to all managed clients.
+        - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L119)
```
Comment (prestwick) [inline release-notes/2024-09/README.md:31]: We should add a line here the mimic the structure and content included when discussion private CA support for `global.trustedCertificateSecrets` (line 22)

**Review by prestwick — CHANGES_REQUESTED** [review]

File: release-notes/2024-09/README.md (hunk 9 above)
Comment (CIakab-NI) [reply release-notes/2024-09/README.md]: Removed the line `- Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on execution pods. These certificates are used when calling SystemLink or other external APIs from the pods.`

File: release-notes/2024-09/README.md (hunk 10 above)
Comment (CIakab-NI) [reply release-notes/2024-09/README.md:23]: I did ask Hem Kumar Rao on Teams to check this input - for some reason I cannot tag his user: hemkumarrao-ni

File: release-notes/2024-09/README.md (hunk 1 above)
Comment (darrenbiel) [reply release-notes/2024-09/README.md]: I updated these with the latest.

File: release-notes/2024-09/README.md (hunk 10 above)
Comment (hemkumarrao-ni) [reply release-notes/2024-09/README.md:23]: This is just a rate-limiting implementation for some APIs(notebook APIs) in the WebApp service. This is very much similar to the rate-limiting done in the Notebook Execution Service mentioned just below in this document. This is not an enterprise helm level change but a service helm level change. 

Reference PR - [Pull Request 787094](https://dev.azure.com/ni/DevCentral/_git/Skyline/pullrequest/787094): WebApp Service | Configure rate limit via Helm chart

We can rephrase it into 
`- WebApp Service
      - Implemented Rate limiting for the notebook APIs`
      
 @prestwick @CIakab-NI @MuthuRajan-S @JD-Robertson

Comment (CIakab-NI) [reply release-notes/2024-09/README.md]: Thanks, @hemkumarrao-ni > I did replace the initial line with the text you had suggested:
- Implemented Rate limiting for the Notebook APIs, in the WebApp Service.

File: release-notes/2024-09/README.md (hunk 12 above)
Comment (CIakab-NI) [reply release-notes/2024-09/README.md]: I did rephrase this as below.
@stefanc18 Stefan Ciuprina, could you help review the 1st and 2nd line below - specifically to the 2nd line could you help suggest the right words. Thanks,

- Systems Management Service support for deploying a custom CA root to all managed clients.
    - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on containers that run Jupyter notebooks. These certificates are used when calling SystemLink or other external APIs from notebooks.
    - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L119)

Comment (stefanc18) [reply release-notes/2024-09/README.md]: The wording looks good to me, but I see that it is mentioning custom certificates on Jupyter notebooks as a subsection of "Systems Management Service support for deploying a custom CA root to all managed clients", which are unrelated. This exact phrase seems to be already written on line 22.

Moreover, in the context of Systems Management Service, custom certificates deployed to managed clients is not done through "global.trustedCertificatesSecrets", but by setting `global.deployApiHostCertificateToSystems` to `true` and providing a single certificate secret at `global.apiHostCertificateSecret`.

File: release-notes/2024-09/README.md (hunk 10 above)
Comment (prestwick) [reply release-notes/2024-09/README.md]: FYI I'm going to adjust the wording of this line. We also lack docs for this. I am not going to hold the release for that.  Bug https://dev.azure.com/ni/DevCentral/_workitems/edit/2605257 filed.

File: release-notes/2024-09/README.md (hunk 12 above)
Comment (prestwick) [reply release-notes/2024-09/README.md]: I reworked all of these for clarity and accuracy

**Review by prestwick — APPROVED** [review]

File: release-notes/2024-09/README.md [hunk 13]
```diff
@@ -0,0 +1,231 @@
...
+    - Search for columns to plot from the axis selection grid.
+    - Color traces by data table and by column metadata.
+
+- Support for [Private Certificate Authorities](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/private-certificate-authorities.html).
+    - Connect to MongoDB and S3 compatible interfaces that use certificates signed by a private authority.
+        - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on containers that connect to MongoDB and S3 compatible interfaces. These certificates are used when connecting to these external dependencies.
+        - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L113)
+    - Call SystemLink APIs from Jupyter Notebook when the SystemLink API endpoint is configured to use a certificate signed by a private authority.
+        - Trusted certificates specified at `global.apiHostCertificateSecret` are now installed on containers that run Jupyter notebooks. These certificates are used when calling SystemLink or other external APIs from notebooks.
```
Comment (stefanc18) [inline release-notes/2024-09/README.md:19]: `global.apiHostCertificateSecret` is used only for the certificate deployed from Systems Management to the systems, not for Jupyter notebooks. JupyterHub & Notebook Executions use the same `global.trustedCertificatesSecrets` that MongoDB and S3 use

**Review by stefanc18 — CHANGES_REQUESTED** [review]

File: release-notes/2024-09/README.md (hunk 13 above)
Comment (prestwick) [reply release-notes/2024-09/README.md]: Good catch. Updated.

File: release-notes/2024-09/README.md [hunk 14]
```diff
@@ -0,0 +1,231 @@
...
+    - Connect to MongoDB and S3 compatible interfaces that use certificates signed by a private authority.
+        - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on containers that connect to MongoDB and S3 compatible interfaces. These certificates are used when connecting to these external dependencies.
+        - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L113)
+    - Call SystemLink APIs from Jupyter Notebook when the SystemLink API endpoint is configured to use a certificate signed by a private authority.
+        - Trusted certificates specified at `global.trustedHostCertificateSecret` are now installed on containers that run Jupyter notebooks. These certificates are used when calling SystemLink or other external APIs from notebooks.
+        - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L119)
```
Comment (stefanc18) [inline release-notes/2024-09/README.md:20]:
```suggestion
        - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L113)
```

**Review by stefanc18 — APPROVED** [review]

File: release-notes/2024-09/README.md [hunk 15]
```diff
@@ -0,0 +1,231 @@
...
+    - Search for columns to plot from the axis selection grid.
+    - Color traces by data table and by column metadata.
+
+- Support for [Private Certificate Authorities](https://www.ni.com/docs/en-US/bundle/systemlink-enterprise/page/private-certificate-authorities.html).
+    - Connect to MongoDB and S3 compatible interfaces that use certificates signed by a private authority.
+        - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on containers that connect to MongoDB and S3 compatible interfaces. These certificates are used when connecting to these external dependencies.
+        - [View this service configuration](https://github.com/ni/install-systemlink-enterprise/blob/2024-09/getting-started/templates/systemlink-values.yaml#L113)
+    - Call SystemLink APIs from Jupyter Notebook when the SystemLink API endpoint is configured to use a certificate signed by a private authority.
+        - Trusted certificates specified at `global.trustedHostCertificateSecret` are now installed on containers that run Jupyter notebooks. These certificates are used when calling SystemLink or other external APIs from notebooks.
```
Comment (stefanc18) [inline release-notes/2024-09/README.md:19]:
```suggestion
        - Trusted certificates specified at `global.trustedCertificatesSecrets` are now installed on containers that run Jupyter notebooks. These certificates are used when calling SystemLink or other external APIs from notebooks.
```

**Review by stefanc18 — APPROVED** [review]

File: release-notes/2024-09/README.md (hunk 14 above)
Comment (prestwick) [reply release-notes/2024-09/README.md]: fixed
